client/components/site-icon/index.jsx

```diff
@@ -70,6 +75,19 @@ const SiteIcon = React.createClass( {
 	}
 } );
 
-export default connect( ( state, { siteId } ) => (
-	siteId ? { site: getSite( state, siteId ) } : {}
-) )( SiteIcon );
+export default connect( ( state, { site, siteId, imgSize } ) => {
+	// Until sites state is completely within Redux, we provide compatibility
+	// in cases where site object is passed to use the icon.img property as URL
+	if ( site ) {
```

Comment:
>Just asking, one could refactor the parent component and provide a "redux" site instead no?

In some cases, yes. But only for sites we know to be in state, which at the moment only includes the currently selected site. Since site icons are shown in the site selector, we can't yet use Redux state there.